### PR TITLE
After zero-rtt has been rejected we should discharge zero-rtt keys.

### LINKED ISSUE #295: We cannot send data after zero-rtt has been rejected
After a zero-rtt has been rejected we try to send data wit zero-rtt keys and we fail the check in this code:
`                    1 => {`
`                        assert!(self.zero_rtt_state != ZeroRttState::Rejected);`
`                        self.zero_rtt_state = ZeroRttState::Sending;`
`                        PacketType::ZeroRTT`

We should discharge zero-rtt keys after it has been rejected.

Also we do not restart stream ids.

### PR DESCRIPTION
**Pull request by ddragana**
Fixes #295

### CODE REVIEW

**Comment by martinthomson** [comment]
We have `Connection::zero_rtt_state` to track this.  In #280 I suggested that we change how we iterate over packet number spaces.  I would prefer to simply remove the 0-RTT keys and then use zero_rtt_state as part of that.

**Comment by ddragana** [comment]
I have moved zero-rtt keys into ZeroRttState.  I have ont change neqo_transport::crypto to use packet number space instead of epoch, because it may conflict with #280 and we should do it in a separate issue.

File: neqo-transport/src/connection.rs [hunk 1]
```diff
@@ -1015,7 +1059,23 @@ impl Connection {
             let mut tokens = Vec::new();
 
             // Ensure we have tx crypto state for this epoch, or skip it.
-            if !matches!(
+            if epoch == 1 {
+                if self.role == Role::Server {
```
[the remark this answers is not in the document]
Comment (ddragana) [reply neqo-transport/src/connection.rs]: No, because of borrowing multiple times. But I made it a bit nice by taking out some part of it.

File: neqo-transport/src/connection.rs [hunk 2]
```diff
@@ -869,12 +870,25 @@ impl Connection {
             debug_assert_eq!(self.role(), Role::Server);
             return None;
         }
-        match self.crypto.obtain_crypto_state(self.role, hdr.epoch) {
-            Ok(CryptoState { rx: Some(rx), .. }) => {
-                let pn_decoder = PacketNumberDecoder::new(largest_acknowledged);
-                decrypt_packet(rx, pn_decoder, &mut hdr, slc).ok()
+        if hdr.epoch == 1 {
```
[the remark this answers is not in the document]
Comment (ddragana) [reply neqo-transport/src/connection.rs]: I have extracted it into a sepparate function.

**Comment by ddragana** [comment]
Rebased and updated.

File: neqo-transport/src/packet.rs [hunk 3]
```diff
@@ -570,6 +570,7 @@ pub fn encode_packet(crypto: &dyn CryptoCtx, hdr: &PacketHdr, body: &[u8]) -> Ve
 #[allow(unused_variables)]
 mod tests {
     use super::*;
+    use neqo_common::matches;
```
Comment (martinthomson) [inline neqo-transport/src/packet.rs]: How is this change necessary?

**Comment by ddragana** [comment]
> Some of this code is starting to look a little scary. But let's get the fix in and then do something about refactoring.

I will open a new issue, write down some thoughts.

File: neqo-transport/src/packet.rs (hunk 3 above)
Comment (ddragana) [reply neqo-transport/src/packet.rs]: I updated rust and it started to complain about ambiguity.
